src-v2/NTS.Domain.Core/Objects/Startlist.cs

```diff
@@ -63,21 +52,16 @@ public void AssignStarts(IEnumerable<Participation> participations)
                         phaseIndex + 1,
                         phases[phaseIndex].Length,
                         participation.Phases.Distance,
-                        phase.StartTime!
+                        phase.StartTime!.DateTime.TimeOfDay
```

Comment:
Using `!` here is safe currently, which is OK, but ideally we should aim to avoid ! unless necessary. Because once the warning is suppressed it cannot be raised again even in valid situations. The diagnostic isn't capable enough to validate that you're filtered away null values. Instead you can do:
```
foreach (var phase in phases)
{
    if (phase.StartTime == null)
    {
        continue;
    }
    var phaseIndex = phases.IndexOf(phase);
    var start = new Start(
        participation.Combination.Name,
        participation.Combination.Number,
        phaseIndex + 1,
        phases[phaseIndex].Length,
        participation.Phases.Distance,
        phase.StartTime.DateTime.TimeOfDay
    );
    Starts.Add(start);
}
```
And avoid suppressing the warning itself.